### PR TITLE
Fix issue #196

### PR DESCRIPTION
**Pull request by jmsantorum**

### CODE REVIEW

File: mongomock/collection.py [hunk 1]
```diff
@@ -1197,6 +1198,84 @@ def group(self, key, condition, initial, reduce, finalize=None):
...
+
+            for doc in out_collection:
+                if key not in doc.keys():
+                    func_field = key.split('__')[1]
+                    func, in_field = func_field.split('_')
+                    out_value = doc.get(in_field)
+
+                    if func in date_operators:
+                        if func == 'dayOfYear':
+                            pass
```
Comment (mdomke) [inline mongomock/collection.py]: How about 
```python
out_value.timetuple().tm_yday
```

File: mongomock/collection.py [hunk 2]
```diff
@@ -1197,6 +1198,84 @@ def group(self, key, condition, initial, reduce, finalize=None):
...
+                        elif func == 'dayOfMonth':
+                            out_value = out_value.day
+                        elif func == 'dayOfWeek':
+                            out_value = out_value.isoweekday()
+                        elif func == 'year':
+                            out_value = out_value.year
+                        elif func == 'month':
+                            out_value = out_value.month
+                        elif func == 'week':
+                            pass
```
Comment (mdomke) [inline mongomock/collection.py]:
```python
out_value.isocalendar()[1]
```

File: mongomock/collection.py [hunk 3]
```diff
@@ -1197,6 +1198,84 @@ def group(self, key, condition, initial, reduce, finalize=None):
...
+                        elif func == 'hour':
+                            out_value = out_value.hour
+                        elif func == 'minute':
+                            out_value = out_value.minute
+                        elif func == 'millisecond':
+                            pass
```
Comment (mdomke) [inline mongomock/collection.py]: The implementation for second would also be trivial or am I missing something here?

**Comment by jmsantorum** [comment]
I've commited the changes that you mention for (dayOfYear, week & millisecond)

File: mongomock/collection.py [hunk 4]
```diff
@@ -1197,6 +1198,84 @@ def group(self, key, condition, initial, reduce, finalize=None):
...
+                        elif func == 'year':
+                            out_value = out_value.year
+                        elif func == 'month':
+                            out_value = out_value.month
+                        elif func == 'week':
+                            out_value = out_value.isocalendar()[1]
+                        elif func == 'hour':
+                            out_value = out_value.hour
+                        elif func == 'minute':
+                            out_value = out_value.minute
```
Comment (mdomke) [inline mongomock/collection.py]: For the sake of completeness please add an implementation for second here also.

**Pull request merged**